### ISSUE TITLE
False positive IDE0032 on nullability difference between field and property

### ISSUE THREAD

**Issue by Ghostbird** [dotnet/roslyn#81176]
**Version Used**: 

.NET 10.0.100 on Debian 13

**Steps to Reproduce**:

[.NET Lab](https://lab.razor.fyi/#3VVPbxtVEG8EQvWqB9Qjp8E9xAFn48ABqVWoqJOAUUujOqUSCMnjt7PeoW_fW715a8eNKnHnxpEL4ovwEYDvwBfgCyCh93bjbEpAcMWSZXt2fjO_-c0fJ9_fSpITZxcOy1TJ7W9vKY0i0JqS8wQAQDx6VrC0nMEjZDPYiebmYXjt7cHEgC9YgM6wrDSBLwiOrR1b4-nMw7swMexBodbAAsYaISOsUHeDfGJXtCQ3BPZQ4nMSCF4EnMd4bMSjUQQ278ZmATZfk_KUpZtoS3SgWocDMLTqIAY79zZ-rU8a6A36nsT3O0-na_FUpmNrxGpKnzn29JANDS5gx9b-T9wvrE9IaKPQy-RlkiRJ6O92CcZ68NZCgVW1hhX7oul5hd6TM0MQDt1hD2xAbEmwwjU4Uo7Qk8Qezj6rtX5COTkyio7OFFWerZnBvI4wBMOKXEAOISOp2BMojVyyWYAv0McwS9Q1gUITOM0JTK11CqcFgSMUayC37gq5MCUBfTcUE9678IygrMUDh4EtyXhAA2w8uRxVmGD0UDm75KzlXjlbkfPrlqc1uyEtznUs9GpQRwsWT4EE-k5UNnA4aVwfG70GzBsnAqwqYMOeUbNgEAWUzQhcbWQYPXImnYVChHwKUxuMjrYFEOaOKYcVm8yuYBXMwH5bWl3avWpN8SsqRSKUBUKRYoNNG2qnLR-FJqjraEOMsuEGolDoKrFZyDejDNBkMAs7NQvmsPaUgVhrmoI7aWYTs0TN2eOKXKy6MxO2XhRh5AIHynNSnpek11AbR6iKqHzQKE2qeq5ZhUER6Wx6cp70xDs2i_uQW3svSXp7e3Aa5-KimVLYWmdxuOd0qcuccuuoOVubEtKk12ZqooZUcPBhiA3374MvnF3FY9Mu39_WNuifOqYs1OYIsxinzXhxtVYY7tpG9rfDXdqkj7c43qyWSFyInaR3nvR6gc1BY7mX9F4mcZW7wHbHOxe8gYTuXS7-7Rtv3ejf3B-lo3R_NHrn1zf29uD3H777EU4nJ3dhYsQH5jbcZpYhrG0d56UWgoxd06xQAozvQM6apDXr9TCuZ_tXETcS4M7dTUfG1uS8qBvJDg5pXi-SpDGm42mBrjpBJ_Q4KikD23yGPrRfYwnpM_bFQzSLGhf0OTkJsr_yOz1xtGRa7VwiThxVzoYhsG66LudWy6B_ePTg6cf9jtcxoa8dySCavtzoaGg16NdCu86KXptdb5-T4Rfk-kPoe1dTf2d41fmYNT1Aoaz9z33V8audJDT-SvljW1asozz_RoSPtLarp0Ywp0EI3akjXOOwRu2-XES73pwemWDswMODkl9EJg9pSXrwF0sa-9fBTCtSnLM6ZFwYK57VRdbrtBxP9z8Y7feH8IQq6_wlKJ3WVeVI5FVBI-K9_4KYHB6NRu9fCzlyzrprO3FU8kaufxI_-D0ijxl6DBd_kKMWiqE-fc3V5ovezZ9---WPn_M3X59tFVvV1tnWN1tbfwI) doesn't actually seem to reproduce the compiler message, but it probably needs more configuration to do so.

**Diagnostic Id**:

`error IDE0032: Use auto property`
Note: Our policy treats all warnings as errors.

**Expected Behavior**:

The nullability of the field does not match the nullability of the property. Hence a conversion to auto-property cannot be performed. It should not be suggested.

**Actual Behavior**:

Compiler says I should convert this to auto property. Auto-fixing this breaks the code, because the `Reset` function assigns `null` to the field.

## Note:

If anyone has a nicer pattern that side-steps the issue, I'd be happy to hear that too.

**Comment by CyrusNajmabadi** [comment]
@Ghostbird please supply a repro in this issue.  Thanks.

**Comment by CyrusNajmabadi** [comment]
Note.  You can do either:

```c#
    public void Reset() {
        foo = null!;
    }
```

Or mark `Foo` as `[AllowNull]`.

**Comment by Ghostbird** [comment]
@CyrusNajmabadi Thank you for the suggestions. I trust you found the "reproduction" then.
